### DISCUSSION TITLE
Absolute vs Relative WER?

### DISCUSSION THREAD

**Discussion by sqrk** [NVIDIA-NeMo/NeMo#5005] [Q&A] [answered]
Hello!

I'm sorry if this isn't the right place to ask this question.

I just want to know the difference between absolute and relative WER. I can't find that information anywhere, everyone just seems to report them but I can't find an actual definition (the wikipedia page for WER doesn't mention anything about absolute or relative).

Thank you in advance.

**Comment by titu1994** [comment, marked as answer]
WER - the score the model gets after inference on a dataset. For example the scores we put on model cards are wer. 

Absolute and Relative wer are when you are doing some comparison between two experiments / models / scores. 

Say model A gets 5% wer and model B gets 6% wer. 
Absolute wer difference = (5 - 6) = 1 Absolute wer difference (degradation). Or in other words, model 2 has worse wer than model 1 (difference being 1 % absolute). 

Relative WER difference = (5 - 6) / 5 = 20 % relative WER difference (degradation). Or in other words, model 2 has a WER that is 20% worse relative to model 1. Note that relative WER always needs a baseline model to compare against,
 (here, 20% worse than model A). 

We had relative WER to indicate improvement when the absolute difference in wer is small - because for sota models, reducing even 0.1% absolute wer is quite significant but doesn't appear very impressive under direct comparison (say 1.6% wer vs 1.5% wer on Librispeech test other, it is actually very good improvement of (1.6-1.5)/1.6 = 6.25% relative improvement over previous sota results).

**Reply by titu1994** [reply]
Also note that here model 2 was worse than model 1, so all the absolute and Relative wer are degredation. If model 2 has 4% wer (ie it is better than model 1), then the absolute and Relative wer are improvements rather than degradation.

You can also keep the sign of the results to indicate the same thing without so many words - + is improvement and - is degradation.

**Reply by sqrk** [reply]
Thank you so much! Makes a lot of sense
